app/main.py

```diff
@@ -2,24 +2,61 @@
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: any, name: str) -> None:
+        self.protected_name = "_" + name
+
+    def __get__(self, instance: any, owner: any) -> int:
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: any, value: int) -> None:
+        if not self.min_amount <= value <= self.max_amount:
+            raise ValueError
+        setattr(instance, self.protected_name, value)
 
 
 class Visitor:
-   pass
+    def __init__(self, name: str, age: int, height: int, weight: int) -> None:
+        self.name = name
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
 class SlideLimitationValidator(ABC):
-    pass
+    def __init__(self, age: int, height: int, weight: int) -> None:
+        self.age = age
+        self.height = height
+        self.weight = weight
 
 
 class ChildrenSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    age = IntegerRange(4, 14)
+    height = IntegerRange(80, 120)
+    weight = IntegerRange(20, 50)
 
 
 class AdultSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    age = IntegerRange(14, 60)
+    height = IntegerRange(120, 220)
+    weight = IntegerRange(50, 120)
 
 
 class Slide:
-    pass
+    def __init__(
+            self,
+            name: str,
+            limitation_class: type
+    ) -> None:
+        self.name = name
+        self.limitations_class = limitation_class
+
+    def can_access(self, visitor: Visitor) -> bool:
+        try:
+            self.limitations_class(visitor.age, visitor.height, visitor.weight)
+        except ValueError:
+            return False
+        return True
```

Comment:
The `can_access` method currently returns `True` or `False`, but it might be more informative to the user if it returned a string explaining why access was denied. This would also make it easier to add more complex rules in the future, as each rule could return its own error message.